raster/r.neighbors/testsuite/test_r_neighbors.py

```diff
@@ -0,0 +1,587 @@
+    raster_mode = "test_neighbors_mode"
+
+    selection = "test_neighbors_selection"
+
+    @classmethod
+    def setUpClass(cls):
+        cls.use_temp_region()
+        cls.runModule("g.region", raster="elevation")
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.del_temp_region()
+        if cls.to_remove:
+            cls.runModule(
+                "g.remove", flags="f", type="raster", name=",".join(cls.to_remove)
+            )
+
+    def test_standard_options(self):
+        """Test output with standard options (size=3, quadratic moving window,
+        standard statistics.
+        """
+        outputs = [
+            self.raster_min,
+            self.raster_perc_90,
+            self.raster_avg,
+            self.raster_max,
+            self.raster_stddev,
+        ]
+        references = [
+            """n=2025000
+null_cells=0
+cells=2025000
+min=55.5787925720215
+max=156.201751708984
+range=100.622959136963
+mean=109.482040656279
+mean_of_abs=109.482040656279
+stddev=20.3075456214408
+variance=412.3964091669
+coeff_var=18.5487459858342
+sum=221701132.328964""",
+            """n=2022154
+null_cells=2846
+cells=2025000
+min=50.5460597991944
+max=153.0166015625
+range=102.470541763306
+mean=100.145484105243
+mean_of_abs=100.145484105243
+stddev=18.2647774894326
+variance=333.602096738483
+coeff_var=18.2382437437099
+sum=202509591.265354""",
+            """n=2025000
+null_cells=0
+cells=2025000
+min=56.0054066975911
+max=156.26158481174
+range=100.256178114149
+mean=110.375489343136
+mean_of_abs=110.375489343136
+stddev=20.302357806975
+variance=412.18573252244
+coeff_var=18.3939006094541
+sum=223510365.91985""",
+            """n=2025000
+null_cells=0
+cells=2025000
+min=56.0526123046875
+max=156.329864501953
+range=100.277252197266
+mean=111.262932473037
+mean_of_abs=111.262932473037
+stddev=20.2987699017146
+variance=412.040059522756
+coeff_var=18.2439644997077
+sum=225307438.2579""",
+            """n=2025000
+null_cells=0
+cells=2025000
+min=0
+max=6.39749880494391
+range=6.39749880494391
+mean=0.576904240112151
+mean_of_abs=0.576904240112151
+stddev=0.440364969041587
+variance=0.193921305958998
+coeff_var=76.3324202567793
+sum=1168231.08622711""",
+            """n=2025000
+null_cells=0
+cells=2025000
+min=233.24821472168
+max=1406.35424804688
+range=1173.1060333252
+mean=992.482567853469
+mean_of_abs=992.482567853469
+stddev=183.486667152836
+variance=33667.3570228558
+coeff_var=18.4876463422103
+sum=2009777199.90327""",
+        ]
+        self.assertModule(
+            "r.neighbors",
+            overwrite=True,
+            input="elevation",
+            output=",".join(outputs),
+            method=["minimum", "perc90", "average", "maximum", "stddev"],
+        )
+        for idx, rmap in enumerate(outputs):
+            self.assertRasterFitsUnivar(
+                raster=rmap, reference=references[idx], precision=1e-5
+            )
+            self.to_remove.append(rmap)
```

Comment:
Potentially, you could skip using to_remove and just try remove all the maps every time, generates warning, but it is not an error. To nicely match the created and deleted maps, you need to add them to to_remove right after creating, not during testing because if an assert fails for any reason, they won't be added. You can probably just `extent()` the list with `outputs`.